scripts/Diffraction/single_crystal/base_sx.py

```diff
@@ -614,6 +618,155 @@ def remove_non_integrated_peaks(peaks):
+                        if log_norm:
+                            im.set_norm(LogNorm())
+                        # plot peak position
+                        ax.plot(0, 0, "xr")
+                        # plot peak representation (circle given extents)
+                        patch = Circle((0, 0), radii[imax] / box_lengths[imax], facecolor="none", edgecolor="r", ls="--")
+                        ax.add_patch(patch)
+                        # add background shell
+                        if np.all(bg_outer_radii > 1e-10):
+                            patch = Circle(
+                                (0, 0), bg_outer_radii[imax] / box_lengths[imax], facecolor="none", edgecolor=3 * [0.7]
+                            )  # outer radius
+                            ax.add_patch(patch)
+                            patch = Circle(
+                                (0, 0), bg_inner_radii[imax] / box_lengths[imax], facecolor="none", edgecolor=3 * [0.7], ls="--"
+                            )  # inner radius
+                            ax.add_patch(patch)
+                        # format axes
+                        ax.set_xlim(-1, 1)
+                        ax.set_ylim(-1, 1)
+                        ax.set_aspect("equal")
+                        ax.set_xlabel(ws_cut.getDimension(dims[0]).name)
+                        ax.set_ylabel(ws_cut.getDimension(dims[1]).name)
+                    fig.suptitle(
+                        f"{ipk} ({','.join(str(np.round(pk.getHKL(), 2))[1:-1].split())})"
+                        f"  $I/\\sigma$={np.round(pk.getIntensityOverSigma(), 2)}\n"
+                        rf"$\lambda$={np.round(pk.getWavelength(), 2)} $\AA$; "
+                        rf"$2\theta={np.round(np.degrees(pk.getScattering()), 1)}^\circ$; "
+                        rf"d={np.round(pk.getDSpacing(), 2)} $\AA$"
+                    )
+                    fig.tight_layout()
+                    pdf.savefig(fig)
+                    close(fig)
+
+                if ws_cut is not None:
+                    mantid.DeleteWorkspace(ws_cut)
+        except OSError:
+            raise RuntimeError(
+                f"OutputFile ({filename}) could not be opened - please check it is not open by "
+                f"another programme and that the user has permission to write to that directory."
+            )
+
+    @staticmethod
+    def _bin_MD_around_peak(wsMD, pk, peak_shape, nbins_max, extent, frame_to_peak_centre_attr):
+        shape_info = json_loads(peak_shape.toJSON())
+        if peak_shape.shapeName().lower() == "spherical":
+            BaseSX.convert_spherical_representation_to_ellipsoid(shape_info)
+        # get radii
+        radii = np.array([shape_info[f"radius{iax}"] for iax in range(3)])
+        bg_inner_radii = np.array([shape_info[f"background_inner_radius{iax}"] for iax in range(3)])
+        bg_outer_radii = np.array([shape_info[f"background_outer_radius{iax}"] for iax in range(3)])
```

Comment:
Why 3, is this the number of axes? Would it be worth adding a constant variable to the top of the file?

Reply:
It's always 3 if the shape is described by an ellipsoid or sphere (in fact always 3 Q-axes in single-crystal diffraction!), but I will make it a variable.